common/utils/src/main/scala/org/apache/spark/SparkThrowableHelper.scala

```diff
@@ -74,6 +74,26 @@ private[spark] object SparkThrowableHelper {
     errorClass.startsWith("INTERNAL_ERROR")
   }
 
+  def isRuntimeUserError(e: Throwable): Boolean = {
```

Comment:
I don't know if "UserError" is right concept here. These are more so Data Dependent errors. I think all exceptions with an error class are user facing.